### ISSUE TITLE
More detail on cosine distance loss function in doc please.

### ISSUE THREAD

**Issue by YagaoDirac** [tensorflow/tfjs#6004]
https://js.tensorflow.org/api/latest/#losses.cosineDistance
I tried to figure out the parameter of axis, but it's not as easy as I expected. By the way, could you please add the formula for all the loss functions in js version in documentation. For people who prefer to know all the details, this could be very helpful. 

Edit: I recently work on a proj to remake the very basic part of tensorflow with both pure typescript and "tf.grads & tf.variablegrad only". If more details are provided, new comers can learn tf in an easier way.

Edit the 2nd time:
```
const axis = 1;
		console.log(axis);
		tf.losses.cosineDistance(
			tf.tensor2d([[1, 1], [1, 1], [1, 1]], [3, 2]),
			tf.tensor2d([[1, 1], [0, 1], [0, 1]], [3, 2]),
			axis).print();
```
axis: 0  result:    -1
axis: 1  result:    -0.333333
I calculated it manually, 
1-(dot([1,1],[1,1])+dot([1,1],[0,1])+dot([1,1],[0,1]))/3 === -1/3
1-(dot([1,1,1],[1,0,0])+dot([1,1,1],[1,1,1]))/2 === -1
Means the structure is [the 1 axis][the 0 axis]. But I remember some of the functions treat the axis in the opposite sequence, for instance, [the 0 axis][the 1 axis][the 2 axis][the 3...]
I really recommend that you add more detail or examples to help people handle these details.

But, what I would like to ask is that, since this function calculates the cosine distance, it should give out result only impacted by the angle. I mean cosine distance should be 1-cos, but your function returns 1-cos*(geometric length of label vector)*(geometric length of predict vector). The 2 vectors are not normalized. Is it intentional? How is this loss function supposed to be used? If normalization is necessary, then what should I do?

Thanks and have a good day.

**Comment by gaikwadrahul8** [comment]
Hi, @YagaoDirac 

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFJs team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TFJs [version](https://github.com/tensorflow/tfjs/releases/tag/tfjs-v4.10.0) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tfjs/blob/master/README.md) to stay up to date with the latest developments which are happening in the Tensorflow.js space.

Thank you for your support and cooperation.
